syntax = "proto3";

option go_package = "github.com/nrf24l01/cp_money_contoller/pb";

package worker;

// Сообщения

message RegisterRequest {
    uint64 id = 1;
}

message RegisterResponse {
    string uuid = 1;
    string key = 2;
    uint64 id = 3;
}

message GetTaskRequest {
    string worker_uuid = 1;
    string worker_key = 2;
}

message GetTaskResponse {
    string uuid = 1;
    string task = 2;
    string payload = 3;
    uint64 unix_time = 4;
}

message ChangeStatusRequest {
    string uuid = 1;
    string payload = 2;
    string logs = 3;
    string status = 4;
    uint64 unix_time = 5;
}

message ChangeStatusResponse {
    bool ok = 1;
}

// Сервис
service WorkerService {
    rpc RegisterWorker(RegisterRequest) returns (RegisterResponse);
    rpc GetTask(GetTaskRequest) returns (GetTaskResponse);
    rpc ChangeStatus(ChangeStatusRequest) returns (ChangeStatusResponse);
}
